gadget/install/partition.go

```diff
@@ -177,42 +177,62 @@ func deviceName(name string, index int) string {
 
 // removeCreatedPartitions removes partitions added during a previous install.
 func removeCreatedPartitions(lv *gadget.LaidOutVolume, dl *gadget.OnDiskVolume) error {
-	indexes := make([]string, 0, len(dl.Structure))
+	sfDiskindexes := make([]string, 0, len(dl.Structure))
+	deletedIndexes := make([]int, 0, len(dl.Structure))
```

Comment:
deletedIndex could be make(map[int]bool,4) instead

Reply:
done